services/pkgreview/package-rating.service.js

```diff
@@ -0,0 +1,99 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { starRating, metric } = require('../text-formatters')
+const { colorScale } = require('../color-formatters')
+const { nonNegativeInteger } = require('../validators')
+const { BaseJsonService } = require('..')
+
+const pkgReviewColor = colorScale([2, 3, 4])
+
+const schema = Joi.object({
+  rating: [
+    Joi.number()
+      .min(1)
+      .max(5)
+      .optional(),
+    Joi.allow(null),
+  ],
+  reviewsCount: nonNegativeInteger,
+}).required()
+
+module.exports = class PkgreviewRating extends BaseJsonService {
+  static get category() {
+    return 'rating'
+  }
+
+  static get route() {
+    return {
+      base: 'pkgreview',
+      pattern: ':format(rating|stars)/:pkgManager/:pkgSlug+',
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'pkgreview.dev Package Ratings',
+        pattern: 'rating/:pkgManager/:pkgSlug+',
+        namedParams: { pkgManager: 'npm', pkgSlug: 'react' },
+        staticPreview: this.render({
+          format: 'rating',
+          rating: 0.7,
+          reviewsCount: 237,
+        }),
+      },
+      {
+        title: 'pkgreview.dev Star Ratings',
+        pattern: 'stars/:pkgManager/:pkgSlug+',
+        namedParams: { pkgManager: 'npm', pkgSlug: 'react' },
+        staticPreview: this.render({
+          format: 'stars',
+          rating: 0.3,
+          reviewsCount: 200,
+        }),
+      },
+    ]
+  }
+
+  static render({ rating, reviewsCount, format }) {
+    const message =
+      format === 'rating'
+        ? `${+parseFloat(rating * 5).toFixed(1)}/5 (${metric(reviewsCount)})`
+        : starRating(rating * 5)
+
+    return {
+      message,
+      label: format,
+      color: pkgReviewColor(rating * 5), // floorCount(rating * 5, 2, 3, 4),
+    }
+  }
+
+  async fetch({ pkgManager, pkgSlug }) {
+    return this._requestJson({
+      schema,
+      url: `https://pkgreview.dev/api/v1/${pkgManager}/${pkgSlug}`,
+      errorMessages: {
+        400: 'package not found',
+        408: 'response timed out',
+        502: 'response timed out',
+        500: 'response timed out',
+      },
+    })
+  }
+
+  transform({ pkgSlug, ...json }) {
+    return {
+      ...json,
+      pkgSlug: encodeURIComponent(pkgSlug),
+    }
+  }
+
```

Comment:
`transform` receives the raw response, knows the details of the response, and extracts/transforms the relevant values. I'm guessing what's in here right now is leftover from when you were embedding the link into the badge, but right now this is basically just injecting a key (`pkgSlug`) to the raw response that's not used anywhere else. 

The only real "transformation" that's happening within this service class is the conversion of raw rating percentage to the 5 star scale (I mentioned this above above in the inline comments on the `render` function) so you could probably get rid of this `transform` altogether and just add that one line in handle (see below) 

```suggestion
```

Reply:
`encodeURIComponent(pkgSlug)` is needed in `fetch` for fetching packages containing `%2f` in the slug.

I have updated the code regarding that in commit 135eaf49d37b10385359b17c937bc1524bf133ab. Please check out if it follows the guidelines. ♥